### ISSUE TITLE
CapacitorSQLite.toJSON() is not implemented on android

### ISSUE THREAD

**Issue by doctorstones** [capacitor-community/sqlite#134]
After updating from Capacitor 2 to 3.0 and update of all relative packages, I've this error on every build.

![capacitor_tojson](https://user-images.githubusercontent.com/3304185/122207234-34c0b380-cea2-11eb-8ffa-6acec8a8b088.png)

Following is my package.json:

"@capacitor-community/sqlite": "^3.0.0-rc.2",
"@capacitor/android": "^3.0.0-rc.4",
"@capacitor/app": "^1.0.1",
"@capacitor/camera": "^1.0.1",
"@capacitor/core": "^3.0.0-rc.4",
[...]
"devDependencies": {
    "@capacitor/cli": "^3.0.0-rc.4",
    "react-sqlite-hook": "^2.0.2"
 },

Every time I build or re-run the app by Android Studio, it fails with the attached screenshot.
This does not impact on the queries itself, calls to plugins go successfully.

I've seen the ionic/react app example, rebuilded everything, cleared everything but nothing seems to work.
I'm encountering this bug only in Android.

**Comment by jepiqueau** [comment]
@doctorstones update capacitor to 3.0.0 do the following
Delete the Android an iOS folder from your app 
npm i --save @capacitor/core@3.0.0
npm i --save @capacitor/android@3.0.0
npm i --save @capacitor/ios@3.0.0
npm I --save-de @capacitor/cli@3.0.0
npm run build
npx cap add android
npx cap add ios
npx cap sync
npm run build
npx cap copy
npx cap open ios
npx cap open android
In Android Studio check that the gradle version is 7.0 if not set it to 7.0
Clean the project
Sync project with gradle files
Build your project and run the app
It should work

**Comment by doctorstones** [comment]
Thank you @jepiqueau ! It works.

**Comment by martinkreitl** [comment]
Hello,

I am currently having this problem too! 

![1625671814489](https://user-images.githubusercontent.com/7662586/124788069-57963300-df49-11eb-8294-e0db5066945f.jpg)
<img width="1769" alt="Bildschirmfoto 2021-07-07 um 15 55 21" src="https://user-images.githubusercontent.com/7662586/124788184-6d0b5d00-df49-11eb-879a-26f84fefec7b.png">

I already updated everything, tried different versions and did all the steps several times.

These are the versions I currently use:
    "@capacitor-community/sqlite": "^3.0.0-rc.2",
    "@capacitor/android": "^3.0.2",
    "@capacitor/core": "^3.0.2",
    "@capacitor/ios": "^3.0.2",
    "@angular/cli": "^12.0.3",

Gradle is 7.0. 

On iOS everything works fine!

**Comment by jepiqueau** [comment]
@martinkreitl which framework are you using ?. I need to have more info on the code can you share it or share a small app replicating the issue

**Comment by martinkreitl** [comment]
Angular.
I will send you the small app as soon as possible.

**Comment by martinkreitl** [comment]
@jepiqueau After I was stripping away all the unnecessary things for the demo app and also changing appId, I tested it on my phone and the demo app unexpectedly worked.

It turns out that deleting the original app of my phone was not enough. I had to delete also all data and cache.
This is what I did then:
settings>>applications>>(application name)>>Storage>> clear cache and clear data

By clearing the application data and the application cache, you also wipe the sql database. This did the trick. Somehow maybe the old version of the database did not work with the new plugin version or something like that.
Then the error did not come back and it works now!
